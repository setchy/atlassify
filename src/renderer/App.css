html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
