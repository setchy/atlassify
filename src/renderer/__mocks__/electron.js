// /**
//  * @typedef {import('../../preload/types.ts').AtlassifyAPI} AtlassifyAPI
//  */

// @ts-ignore
window.Notification = function (title) {
  this.title = title;

  return {
    onclick: jest.fn(),
  };
};

// @ts-ignore
window.Audio = class Audio {
  constructor(path) {
    this.path = path;
  }

  play() {}
};

// @ts-ignore
window.localStorage = {
  store: {},
  getItem: function (key) {
    return this.store[key];
  },
  setItem: function (key, item) {
    this.store[key] = item;
  },
  removeItem: jest.fn(),
};

window.alert = jest.fn();

// /** @type {AtlassifyAPI} */
// const mockAtlassifyAPI = {
// 	openExternalLink: jest.fn(),
// 	decryptValue: () => Promise.resolve("decrypted"),
// 	encryptValue: () => Promise.resolve("encrypted"),
// 	platform: {
// 		isLinux: () => false,
// 		isMacOS: () => true,
// 		isWindows: () => false,
// 	},
// 	app: {
// 		hide: jest.fn(),
// 		show: jest.fn(),
// 		quit: jest.fn(),
// 		version: () => Promise.resolve("0.0.1"),
// 	},
// 	zoom: {
// 		getLevel: jest.fn(),
// 		setLevel: jest.fn(),
// 	},
// 	tray: {
// 		updateIcon: jest.fn(),
// 		updateTitle: jest.fn(),
// 		useAlternateIdleIcon: jest.fn(),
// 	},
// 	twemojiDirectory: jest.fn(),
// };

// window.atlassify = mockAtlassifyAPI;

// window.atlassify.decryptValue = Promise.resolve("decrypted");
// window.atlassify.twemojiDirectory = Promise.resolve("/tmp/emoji");

window.atlassify.app = {
  version: Promise.resolve('0.0.1'),
};
