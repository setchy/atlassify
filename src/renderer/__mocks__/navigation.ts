const mockNavigate = vi.fn();

vi.mock('react-router-dom', async () => ({
  ...(await vi.importActual('react-router-dom')),
  useNavigate: () => mockNavigate,
}));

export { mockNavigate };
