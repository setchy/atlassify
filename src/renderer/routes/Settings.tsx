import type { FC } from 'react';

import { Box, Stack } from '@atlaskit/primitives';

import { Contents } from '../components/primitives/Contents';
import { Header } from '../components/primitives/Header';
import { Page } from '../components/primitives/Page';
import { AppearanceSettings } from '../components/settings/AppearanceSettings';
import { NotificationSettings } from '../components/settings/NotificationSettings';
import { SettingsFooter } from '../components/settings/SettingsFooter';
import { SettingsReset } from '../components/settings/SettingsReset';
import { SystemSettings } from '../components/settings/SystemSettings';

// FIXME #90 InlineMessage findDOMNode errors
export const SettingsRoute: FC = () => {
  return (
    <Page id="settings">
      <Header fetchOnBack>Settings</Header>

      <Contents>
        <Box paddingInline="space.400" paddingBlockEnd="space.200">
          <Stack space="space.200">
            <AppearanceSettings />
            <NotificationSettings />
            <SystemSettings />
            <SettingsReset />
          </Stack>
        </Box>
      </Contents>

      <SettingsFooter />
    </Page>
  );
};
