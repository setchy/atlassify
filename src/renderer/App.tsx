import { useContext } from 'react';
import {
  Navigate,
  Route,
  HashRouter as Router,
  Routes,
  useLocation,
} from 'react-router-dom';

import { AppContext, AppProvider } from './context/App';
import { AccountsRoute } from './routes/Accounts';
import { FiltersRoute } from './routes/Filters';
import { LandingRoute } from './routes/Landing';
import { LoginRoute } from './routes/Login';
import { NotificationsRoute } from './routes/Notifications';
import { SettingsRoute } from './routes/Settings';

import './App.css';
import { AppLayout } from './components/layout/AppLayout';

function RequireAuth({ children }) {
  const { isLoggedIn } = useContext(AppContext);
  const location = useLocation();

  return isLoggedIn ? (
    children
  ) : (
    <Navigate to="/landing" replace state={{ from: location }} />
  );
}

export const App = () => {
  return (
    <AppProvider>
      <Router>
        <AppLayout>
          <Routes>
            <Route
              path="/"
              element={
                <RequireAuth>
                  <NotificationsRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/filters"
              element={
                <RequireAuth>
                  <FiltersRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/settings"
              element={
                <RequireAuth>
                  <SettingsRoute />
                </RequireAuth>
              }
            />
            <Route
              path="/accounts"
              element={
                <RequireAuth>
                  <AccountsRoute />
                </RequireAuth>
              }
            />
            <Route path="/landing" element={<LandingRoute />} />
            <Route path="/login" element={<LoginRoute />} />
          </Routes>
        </AppLayout>
      </Router>
    </AppProvider>
  );
};
