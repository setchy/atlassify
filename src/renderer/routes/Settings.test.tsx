import { act, fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { mockNavigate } from '../__mocks__/navigation';
import { mockAuth, mockSettings } from '../__mocks__/state';

import { AppContext } from '../context/App';
import { SettingsRoute } from './Settings';

describe('renderer/routes/Settings.tsx', () => {
  const fetchNotifications = vi.fn();

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('should render itself & its children', async () => {
    await act(async () => {
      render(
        <AppContext.Provider value={{ auth: mockAuth, settings: mockSettings }}>
          <MemoryRouter>
            <SettingsRoute />
          </MemoryRouter>
        </AppContext.Provider>,
      );
    });

    expect(screen.getByTestId('settings')).toMatchSnapshot();
  });

  it('should go back by pressing the icon', async () => {
    await act(async () => {
      render(
        <AppContext.Provider
          value={{
            auth: mockAuth,
            settings: mockSettings,
            fetchNotifications,
          }}
        >
          <MemoryRouter>
            <SettingsRoute />
          </MemoryRouter>
        </AppContext.Provider>,
      );
    });

    fireEvent.click(screen.getByTestId('header-nav-back'));

    expect(fetchNotifications).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenNthCalledWith(1, -1);
  });
});
