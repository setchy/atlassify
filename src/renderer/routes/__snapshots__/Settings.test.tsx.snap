// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/routes/Settings.tsx should render itself & its children 1`] = `
<div
  class="flex h-screen flex-col"
  data-testid="settings"
>
  <div
    class="css-18bes35"
  >
    <div
      class="css-y272c0"
    >
      <div
        role="presentation"
      >
        <button
          class="css-1hmr5rx"
          data-testid="header-nav-back"
          type="button"
        >
          <span
            class="css-1spmf3f"
          >
            <span
              aria-hidden="true"
              class="css-snhnyn"
              data-vc="icon-undefined"
              style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
            >
              <svg
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.207 5.793a1 1 0 0 1 0 1.414L8.414 11H18a1 1 0 1 1 0 2H8.414l3.793 3.793a1 1 0 1 1-1.415 1.414l-5.499-5.5A1 1 0 0 1 5 12.028v-.057a1 1 0 0 1 .293-.678l5.499-5.5a1 1 0 0 1 1.415 0"
                  fill="currentcolor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="css-b5o75w"
            >
              Go Back
            </span>
          </span>
        </button>
      </div>
      <h3
        class="css-rex7s6"
      >
        Settings
      </h3>
    </div>
  </div>
  <div
    class="flex flex-col flex-grow overflow-x-auto"
  >
    <div
      class="css-8mzmlv"
    >
      <div
        class="css-1ub26k9"
      >
        <div
          class="css-t8lj6l"
        >
          <h4
            class="css-1skdotg"
          >
            Appearance
          </h4>
          <div
            class="css-gokkc4"
          >
            <span
              class="css-19g3t4k"
              id="theme-label"
            >
              Theme:
            </span>
            <div
              aria-labelledby="theme-label"
              role="radiogroup"
            >
              <label
                class="css-exvs2c"
                data-testid="theme-system--radio-label"
              >
                <input
                  class="css-t4vzni"
                  data-testid="theme-system--radio-input"
                  name="theme"
                  tabindex="0"
                  type="radio"
                  value="SYSTEM"
                />
                <span
                  class="css-b9c6gp"
                >
                  System
                </span>
              </label>
              <label
                class="css-exvs2c"
                data-testid="theme-light--radio-label"
              >
                <input
                  checked=""
                  class="css-t4vzni"
                  data-testid="theme-light--radio-input"
                  name="theme"
                  tabindex="0"
                  type="radio"
                  value="LIGHT"
                />
                <span
                  class="css-b9c6gp"
                >
                  Light
                </span>
              </label>
              <label
                class="css-exvs2c"
                data-testid="theme-dark--radio-label"
              >
                <input
                  class="css-t4vzni"
                  data-testid="theme-dark--radio-input"
                  name="theme"
                  tabindex="0"
                  type="radio"
                  value="DARK"
                />
                <span
                  class="css-b9c6gp"
                >
                  Dark
                </span>
              </label>
            </div>
          </div>
          <div
            class="css-gokkc4"
          >
            <span
              class="css-19g3t4k"
              id="theme-label"
            >
              Zoom:
            </span>
            <div
              class="css-mrgpd6"
            >
              <div
                role="presentation"
              >
                <button
                  class="css-1h9zni"
                  data-testid="settings-zoom-out"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-snhnyn"
                      data-vc="icon-undefined"
                      style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                    >
                      <svg
                        height="24"
                        role="presentation"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="m16.436 15.085 3.94 4.01a1 1 0 0 1-1.425 1.402l-3.938-4.006a7.5 7.5 0 1 1 1.423-1.406M13.5 11.5h-6a1 1 0 0 1 0-2h6a1 1 0 0 1 0 2"
                          fill="currentcolor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Zoom Out
                    </span>
                  </span>
                </button>
              </div>
              <span
                class="css-1wkl091"
              >
                <span
                  class="css-f9etm4"
                >
                  100
                  %
                </span>
              </span>
              <div
                role="presentation"
              >
                <button
                  class="css-1h9zni"
                  data-testid="settings-zoom-in"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-snhnyn"
                      data-vc="icon-undefined"
                      style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                    >
                      <svg
                        height="24"
                        role="presentation"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="m16.436 15.085 3.94 4.01a1 1 0 0 1-1.425 1.402l-3.938-4.006a7.5 7.5 0 1 1 1.423-1.406M13.5 11.5a1 1 0 0 0 0-2h-2v-2a1 1 0 0 0-2 0v2h-2a1 1 0 0 0 0 2h2v2a1 1 0 0 0 2 0v-2z"
                          fill="currentcolor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Zoom In
                    </span>
                  </span>
                </button>
              </div>
              <div
                role="presentation"
              >
                <button
                  class="css-1h9zni"
                  data-testid="settings-zoom-reset"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-snhnyn"
                      data-vc="icon-undefined"
                      style="--icon-primary-color: var(--ds-icon-accent-red); --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                    >
                      <svg
                        height="24"
                        role="presentation"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <g
                          fill-rule="evenodd"
                        >
                          <circle
                            cx="12"
                            cy="12"
                            fill="currentcolor"
                            r="9"
                          />
                          <path
                            d="M16.155 14.493a1.174 1.174 0 1 1-1.662 1.663L12 13.662l-2.494 2.494a1.17 1.17 0 0 1-1.662 0 1.176 1.176 0 0 1 0-1.663L10.337 12 7.844 9.507a1.176 1.176 0 0 1 1.662-1.662L12 10.338l2.493-2.493a1.174 1.174 0 1 1 1.662 1.662L13.662 12z"
                            fill="inherit"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Reset Zoom
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-t8lj6l"
        >
          <h4
            class="css-1skdotg"
          >
            Notifications
          </h4>
          <label
            class="css-k8x29e"
          >
            <input
              checked=""
              class="css-f3myyk"
              name="markAsReadOnOpen"
              tabindex="0"
              type="checkbox"
              value=""
            />
            <svg
              class="css-1dr55qw"
              role="presentation"
              style="fill: var(--checkbox-tick-color);"
              viewBox="0 0 24 24"
            >
              <g
                fill-rule="evenodd"
              >
                <rect
                  fill="currentColor"
                  height="12"
                  rx="2"
                  width="12"
                  x="6"
                  y="6"
                />
                <path
                  d="M9.707 11.293a1 1 0 1 0-1.414 1.414l2 2a1 1 0 0 0 1.414 0l4-4a1 1 0 1 0-1.414-1.414L11 12.586l-1.293-1.293z"
                  fill="inherit"
                />
              </g>
            </svg>
            <span
              class="css-1odt9zb"
            >
              Mark as read on open
            </span>
          </label>
          <div
            class="css-gokkc4"
          >
            <label
              class="css-k8x29e"
            >
              <input
                class="css-f3myyk"
                name="groupNotificationsByProductAlphabetically"
                tabindex="0"
                type="checkbox"
                value=""
              />
              <svg
                class="css-1dr55qw"
                role="presentation"
                style="fill: var(--checkbox-tick-color);"
                viewBox="0 0 24 24"
              >
                <g
                  fill-rule="evenodd"
                >
                  <rect
                    fill="currentColor"
                    height="12"
                    rx="2"
                    width="12"
                    x="6"
                    y="6"
                  />
                </g>
              </svg>
              <span
                class="css-1odt9zb"
              >
                Group product notifications alphabetically
              </span>
            </label>
            <div
              class="css-ssyr0c"
              style="--icon-accent-color: var(--ds-icon-discovery, #8777D9);"
            >
              <button
                aria-expanded="false"
                class="css-1au32bq"
                data-ds--inline-message--button="true"
                tabindex="0"
                type="button"
              >
                <span
                  class="css-1gd7hga"
                >
                  <div
                    class="css-bsfd3j"
                  >
                    <span
                      class="css-q0v5v7"
                      data-ds--inline-message--icon="true"
                      style="--icon-color: var(--ds-icon-discovery, #6554C0);"
                    >
                      <span
                        aria-label="info inline message"
                        class="css-1afrefi"
                        data-testid="inline-message-icon"
                        data-vc="icon-inline-message-icon"
                        role="img"
                        style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                      >
                        <svg
                          height="24"
                          role="presentation"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <g
                            fill-rule="evenodd"
                          >
                            <path
                              d="M2 12c0 5.523 4.477 10 10 10s10-4.477 10-10S17.523 2 12 2 2 6.477 2 12"
                              fill="currentcolor"
                            />
                            <rect
                              fill="inherit"
                              height="7"
                              rx="1"
                              width="2"
                              x="11"
                              y="10"
                            />
                            <circle
                              cx="12"
                              cy="8"
                              fill="inherit"
                              r="1"
                            />
                          </g>
                        </svg>
                      </span>
                    </span>
                    <div
                      class="css-gokkc4"
                    />
                  </div>
                </span>
              </button>
            </div>
          </div>
          <div
            class="css-gokkc4"
          >
            <label
              class="css-k8x29e"
            >
              <input
                class="css-f3myyk"
                name="delayNotificationState"
                tabindex="0"
                type="checkbox"
                value=""
              />
              <svg
                class="css-1dr55qw"
                role="presentation"
                style="fill: var(--checkbox-tick-color);"
                viewBox="0 0 24 24"
              >
                <g
                  fill-rule="evenodd"
                >
                  <rect
                    fill="currentColor"
                    height="12"
                    rx="2"
                    width="12"
                    x="6"
                    y="6"
                  />
                </g>
              </svg>
              <span
                class="css-1odt9zb"
              >
                Delay notification state
              </span>
            </label>
            <div
              class="css-ssyr0c"
              style="--icon-accent-color: var(--ds-icon-discovery, #8777D9);"
            >
              <button
                aria-expanded="false"
                class="css-1au32bq"
                data-ds--inline-message--button="true"
                tabindex="0"
                type="button"
              >
                <span
                  class="css-1gd7hga"
                >
                  <div
                    class="css-bsfd3j"
                  >
                    <span
                      class="css-q0v5v7"
                      data-ds--inline-message--icon="true"
                      style="--icon-color: var(--ds-icon-discovery, #6554C0);"
                    >
                      <span
                        aria-label="info inline message"
                        class="css-1afrefi"
                        data-testid="inline-message-icon"
                        data-vc="icon-inline-message-icon"
                        role="img"
                        style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                      >
                        <svg
                          height="24"
                          role="presentation"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <g
                            fill-rule="evenodd"
                          >
                            <path
                              d="M2 12c0 5.523 4.477 10 10 10s10-4.477 10-10S17.523 2 12 2 2 6.477 2 12"
                              fill="currentcolor"
                            />
                            <rect
                              fill="inherit"
                              height="7"
                              rx="1"
                              width="2"
                              x="11"
                              y="10"
                            />
                            <circle
                              cx="12"
                              cy="8"
                              fill="inherit"
                              r="1"
                            />
                          </g>
                        </svg>
                      </span>
                    </span>
                    <div
                      class="css-gokkc4"
                    />
                  </div>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-t8lj6l"
        >
          <h4
            class="css-1skdotg"
          >
            System
          </h4>
          <div
            class="css-gokkc4"
          >
            <span
              class="css-19g3t4k"
              id="openLinks-label"
            >
              Open Links:
            </span>
            <div
              aria-labelledby="openLinks-label"
              role="radiogroup"
            >
              <label
                class="css-exvs2c"
              >
                <input
                  checked=""
                  class="css-t4vzni"
                  name="openLinks"
                  tabindex="0"
                  type="radio"
                  value="FOREGROUND"
                />
                <span
                  class="css-b9c6gp"
                >
                  Foreground
                </span>
              </label>
              <label
                class="css-exvs2c"
              >
                <input
                  class="css-t4vzni"
                  name="openLinks"
                  tabindex="0"
                  type="radio"
                  value="BACKGROUND"
                />
                <span
                  class="css-b9c6gp"
                >
                  Background
                </span>
              </label>
            </div>
          </div>
          <div
            class="css-gokkc4"
          >
            <label
              class="css-k8x29e"
            >
              <input
                checked=""
                class="css-f3myyk"
                name="keyboardShortcutEnabled"
                tabindex="0"
                type="checkbox"
                value=""
              />
              <svg
                class="css-1dr55qw"
                role="presentation"
                style="fill: var(--checkbox-tick-color);"
                viewBox="0 0 24 24"
              >
                <g
                  fill-rule="evenodd"
                >
                  <rect
                    fill="currentColor"
                    height="12"
                    rx="2"
                    width="12"
                    x="6"
                    y="6"
                  />
                  <path
                    d="M9.707 11.293a1 1 0 1 0-1.414 1.414l2 2a1 1 0 0 0 1.414 0l4-4a1 1 0 1 0-1.414-1.414L11 12.586l-1.293-1.293z"
                    fill="inherit"
                  />
                </g>
              </svg>
              <span
                class="css-1odt9zb"
              >
                Enable keyboard shortcut
              </span>
            </label>
            <div
              class="css-ssyr0c"
              style="--icon-accent-color: var(--ds-icon-discovery, #8777D9);"
            >
              <button
                aria-expanded="false"
                class="css-1au32bq"
                data-ds--inline-message--button="true"
                tabindex="0"
                type="button"
              >
                <span
                  class="css-1gd7hga"
                >
                  <div
                    class="css-bsfd3j"
                  >
                    <span
                      class="css-q0v5v7"
                      data-ds--inline-message--icon="true"
                      style="--icon-color: var(--ds-icon-discovery, #6554C0);"
                    >
                      <span
                        aria-label="info inline message"
                        class="css-1afrefi"
                        data-testid="inline-message-icon"
                        data-vc="icon-inline-message-icon"
                        role="img"
                        style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                      >
                        <svg
                          height="24"
                          role="presentation"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <g
                            fill-rule="evenodd"
                          >
                            <path
                              d="M2 12c0 5.523 4.477 10 10 10s10-4.477 10-10S17.523 2 12 2 2 6.477 2 12"
                              fill="currentcolor"
                            />
                            <rect
                              fill="inherit"
                              height="7"
                              rx="1"
                              width="2"
                              x="11"
                              y="10"
                            />
                            <circle
                              cx="12"
                              cy="8"
                              fill="inherit"
                              r="1"
                            />
                          </g>
                        </svg>
                      </span>
                    </span>
                    <div
                      class="css-gokkc4"
                    />
                  </div>
                </span>
              </button>
            </div>
          </div>
          <label
            class="css-k8x29e"
          >
            <input
              checked=""
              class="css-f3myyk"
              name="showNotificationsCountInTray"
              tabindex="0"
              type="checkbox"
              value=""
            />
            <svg
              class="css-1dr55qw"
              role="presentation"
              style="fill: var(--checkbox-tick-color);"
              viewBox="0 0 24 24"
            >
              <g
                fill-rule="evenodd"
              >
                <rect
                  fill="currentColor"
                  height="12"
                  rx="2"
                  width="12"
                  x="6"
                  y="6"
                />
                <path
                  d="M9.707 11.293a1 1 0 1 0-1.414 1.414l2 2a1 1 0 0 0 1.414 0l4-4a1 1 0 1 0-1.414-1.414L11 12.586l-1.293-1.293z"
                  fill="inherit"
                />
              </g>
            </svg>
            <span
              class="css-1odt9zb"
            >
              Show notifications count in tray
            </span>
          </label>
          <label
            class="css-k8x29e"
          >
            <input
              checked=""
              class="css-f3myyk"
              name="showNotifications"
              tabindex="0"
              type="checkbox"
              value=""
            />
            <svg
              class="css-1dr55qw"
              role="presentation"
              style="fill: var(--checkbox-tick-color);"
              viewBox="0 0 24 24"
            >
              <g
                fill-rule="evenodd"
              >
                <rect
                  fill="currentColor"
                  height="12"
                  rx="2"
                  width="12"
                  x="6"
                  y="6"
                />
                <path
                  d="M9.707 11.293a1 1 0 1 0-1.414 1.414l2 2a1 1 0 0 0 1.414 0l4-4a1 1 0 1 0-1.414-1.414L11 12.586l-1.293-1.293z"
                  fill="inherit"
                />
              </g>
            </svg>
            <span
              class="css-1odt9zb"
            >
              Show system notifications
            </span>
          </label>
          <label
            class="css-k8x29e"
          >
            <input
              checked=""
              class="css-f3myyk"
              name="playSoundNewNotifications"
              tabindex="0"
              type="checkbox"
              value=""
            />
            <svg
              class="css-1dr55qw"
              role="presentation"
              style="fill: var(--checkbox-tick-color);"
              viewBox="0 0 24 24"
            >
              <g
                fill-rule="evenodd"
              >
                <rect
                  fill="currentColor"
                  height="12"
                  rx="2"
                  width="12"
                  x="6"
                  y="6"
                />
                <path
                  d="M9.707 11.293a1 1 0 1 0-1.414 1.414l2 2a1 1 0 0 0 1.414 0l4-4a1 1 0 1 0-1.414-1.414L11 12.586l-1.293-1.293z"
                  fill="inherit"
                />
              </g>
            </svg>
            <span
              class="css-1odt9zb"
            >
              Play sound for new notifications
            </span>
          </label>
          <div
            class="css-gokkc4"
          >
            <label
              class="css-k8x29e"
            >
              <input
                class="css-f3myyk"
                name="useAlternateIdleIcon"
                tabindex="0"
                type="checkbox"
                value=""
              />
              <svg
                class="css-1dr55qw"
                role="presentation"
                style="fill: var(--checkbox-tick-color);"
                viewBox="0 0 24 24"
              >
                <g
                  fill-rule="evenodd"
                >
                  <rect
                    fill="currentColor"
                    height="12"
                    rx="2"
                    width="12"
                    x="6"
                    y="6"
                  />
                </g>
              </svg>
              <span
                class="css-1odt9zb"
              >
                Use alternate idle icon
              </span>
            </label>
            <div
              class="css-ssyr0c"
              style="--icon-accent-color: var(--ds-icon-discovery, #8777D9);"
            >
              <button
                aria-expanded="false"
                class="css-1au32bq"
                data-ds--inline-message--button="true"
                tabindex="0"
                type="button"
              >
                <span
                  class="css-1gd7hga"
                >
                  <div
                    class="css-bsfd3j"
                  >
                    <span
                      class="css-q0v5v7"
                      data-ds--inline-message--icon="true"
                      style="--icon-color: var(--ds-icon-discovery, #6554C0);"
                    >
                      <span
                        aria-label="info inline message"
                        class="css-1afrefi"
                        data-testid="inline-message-icon"
                        data-vc="icon-inline-message-icon"
                        role="img"
                        style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                      >
                        <svg
                          height="24"
                          role="presentation"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <g
                            fill-rule="evenodd"
                          >
                            <path
                              d="M2 12c0 5.523 4.477 10 10 10s10-4.477 10-10S17.523 2 12 2 2 6.477 2 12"
                              fill="currentcolor"
                            />
                            <rect
                              fill="inherit"
                              height="7"
                              rx="1"
                              width="2"
                              x="11"
                              y="10"
                            />
                            <circle
                              cx="12"
                              cy="8"
                              fill="inherit"
                              r="1"
                            />
                          </g>
                        </svg>
                      </span>
                    </span>
                    <div
                      class="css-gokkc4"
                    />
                  </div>
                </span>
              </button>
            </div>
          </div>
          <label
            class="css-k8x29e"
          >
            <input
              checked=""
              class="css-f3myyk"
              name="openAtStartUp"
              tabindex="0"
              type="checkbox"
              value=""
            />
            <svg
              class="css-1dr55qw"
              role="presentation"
              style="fill: var(--checkbox-tick-color);"
              viewBox="0 0 24 24"
            >
              <g
                fill-rule="evenodd"
              >
                <rect
                  fill="currentColor"
                  height="12"
                  rx="2"
                  width="12"
                  x="6"
                  y="6"
                />
                <path
                  d="M9.707 11.293a1 1 0 1 0-1.414 1.414l2 2a1 1 0 0 0 1.414 0l4-4a1 1 0 1 0-1.414-1.414L11 12.586l-1.293-1.293z"
                  fill="inherit"
                />
              </g>
            </svg>
            <span
              class="css-1odt9zb"
            >
              Open at startup
            </span>
          </label>
        </div>
        <div
          class="css-ze0mo2"
        >
          <button
            aria-haspopup="dialog"
            class="css-hyamds"
            data-testid="settings-reset-defaults"
            type="button"
          >
            <span
              class="css-1nqby2s"
            >
              Reset Settings
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-1e13sq9"
  >
    <div
      class="css-1npayr"
    >
      <div
        class="css-uwvn2m"
      >
        <div
          role="presentation"
        >
          <button
            class="css-1aearyh"
            data-testid="settings-release-notes"
            title="View Atlassify release notes"
            type="button"
          >
            <span
              class="css-1nqby2s"
            >
              Atlassify 
              v0.0.1
            </span>
          </button>
        </div>
        <div
          class="css-srvzt9"
        >
          <div
            role="presentation"
          >
            <button
              class="css-1hmr5rx"
              data-testid="settings-accounts"
              type="button"
            >
              <span
                class="css-1spmf3f"
              >
                <span
                  aria-hidden="true"
                  class="css-snhnyn"
                  data-vc="icon-undefined"
                  style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                >
                  <svg
                    height="24"
                    role="presentation"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <g
                      fill="currentcolor"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M8.13 10H4a2 2 0 0 0-2 2v3.73c0 1.31 1.87 2.05 4 2.23V15a3 3 0 0 1 3-3h.35a4 4 0 0 1-1.23-2m7.75 0H20c1.11 0 2 .89 2 2v3.73c0 1.31-1.87 2.05-4 2.23V15a3 3 0 0 0-3-3h-.35a4 4 0 0 0 1.23-2M9.97 5.55a3 3 0 1 0-1.96 3.27 4 4 0 0 1 1.96-3.27M16 8.83a3 3 0 1 0-1.96-3.28A4 4 0 0 1 16 8.83"
                      />
                      <path
                        d="M7 15c0-1.105.887-2 2-2h6c1.105 0 2 .885 2 2v3.73c0 3.027-10 3.027-10 0z"
                      />
                      <circle
                        cx="12"
                        cy="9"
                        r="3"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  class="css-b5o75w"
                >
                  Accounts
                </span>
              </span>
            </button>
          </div>
          <div
            role="presentation"
          >
            <button
              class="css-1hmr5rx"
              data-testid="settings-quit"
              type="button"
            >
              <span
                class="css-1spmf3f"
              >
                <span
                  aria-hidden="true"
                  class="css-snhnyn"
                  data-vc="icon-undefined"
                  style="--icon-primary-color: currentColor; --icon-secondary-color: var(--ds-surface, #FFFFFF);"
                >
                  <svg
                    height="24"
                    role="presentation"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <g
                      fill-rule="evenodd"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        fill="currentcolor"
                        r="10"
                      />
                      <path
                        d="m13.477 9.113-4.36 4.386a1 1 0 1 0 1.418 1.41l4.36-4.386a1 1 0 0 0-1.418-1.41"
                        fill="inherit"
                      />
                      <path
                        d="m9.084 10.501 4.358 4.377a1 1 0 1 0 1.418-1.411L10.5 9.09a1 1 0 0 0-1.417 1.411"
                        fill="inherit"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  class="css-b5o75w"
                >
                  Quit Atlassify
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
